### PR TITLE
ci: add repo CI and a moving major tag for v1 releases

### LINKED ISSUE #41: No lint runs against this repo's own workflow files
All six workflows here are `workflow_call`-only, so none of them trigger on this
repo's own pushes or pull requests. There is no test suite, no `run` script and
no lefthook config. A YAML syntax error or a bad `${{ }}` expression is caught
by the first consumer that calls it, after merge to `main` — which is what every
consumer pins.

`translation-status.yml` raises the exposure: it carries the most embedded shell
of the six, with a PATCH/POST branch, `PIPESTATUS` handling and an
outcome-conditional step.

`actionlint` covers this cheaply and runs `shellcheck` over `run:` blocks. It
was run by hand against `translation-status.yml` before merge; nothing makes
that repeatable. Run it on `pull_request` over `.github/workflows/` and
`examples/`.

This will not catch semantic mistakes — the `if: steps.anchors.outcome ==
'failure'` condition that skipped when the gate failed was valid YAML and a
valid expression. It catches the syntax class, which is currently caught by
nobody.

### LINKED ISSUE #50: v1 Unit 1: Repo foundation — CI, release process, AGENTS.md
Part of the shared-workflows v1 plan (tracking issue: [issue 49](https://github.com/halos-org/shared-workflows/issues/49)).

**Goal:** The repo can lint its own workflows and cut `v1.0.0-rc.N`, `v1.0.0`, and move `v1`.

**Requirements:** R3

**Dependencies:** None

**Files:**
- Create: `.github/workflows/lint.yml` (actionlint on PRs; addresses [issue 41](https://github.com/halos-org/shared-workflows/issues/41))
- Create: `.github/workflows/move-major-tag.yml` (on full release published, move `vN`)
- Create: `AGENTS.md`
- Modify: `README.md` (versioning section)

**Approach:**
- Lint runs on this repo's PRs only; it is not `workflow_call`.
- Releases of this repo are created by hand (`vX.Y.Z` or `vX.Y.Z-rc.N`). Major tag moves only for non-prerelease `vX.Y.Z` releases; the workflow needs `contents: write`.
- Before writing the workflow, check the repo's rulesets allow force-updating `v*` tags.
- rc tags are never deleted; a canary left on an rc keeps working.
- AGENTS.md states the rules: legacy files frozen until removal, breaking changes need a new major, job names are internal.
- Add the lint check to this repo's required checks (repo setting; approved as part of R4 work).
- Disable the residual `probe.yml` and `guardprobe.yml` workflow registrations.

**Patterns to follow:** actions/toolkit major-tag process.

**Test scenarios:**
- Happy path: PR with valid workflow YAML → lint passes.
- Error path: PR with an undefined input reference → lint fails.
- Happy path: publish release `v1.0.0` → `v1` points at its commit.
- Edge case: publish pre-release `v1.0.0-rc.1` → `v1` does not move.
- Edge case: publish `v1.1.0` after `v1.0.0` → `v1` moves forward, `v1.0.0` unchanged.

**Verification:** A throwaway release on a fork or test tag moves the major tag as specified; lint check reports on a PR.

### PR DESCRIPTION
**Pull request by mairas**
## Summary

First step of the v1 interface plan ([issue 49](https://github.com/halos-org/shared-workflows/issues/49)): this repository gets its own CI and a release process that maintains a moving major tag, so callers can pin `@v1` instead of `@main`.

## Changes

- **Major tag.** `move-major-tag.yml` runs on a published release and calls `scripts/move-major-tag.sh`. `vX` moves only when the release is a full `vX.Y.Z` and is the highest `vX.*.*` release. A patch for an older minor leaves `vX` on the newest release. Pre-releases (`vX.Y.Z-rc.N`) never move it, and any other tag name is an error. No tag ruleset exists on this repo, so the job token can force-push the tag.
- **CI.** `ci.yml` runs actionlint 1.7.12 over the workflows and `examples/`, shellcheck over `scripts/` and `tests/`, and every `tests/*.test.sh`. A `status` job aggregates the jobs, so branch protection can require one name that job renames do not change.
- **Docs.** `AGENTS.md` (with a `CLAUDE.md` include) records the interface boundary, the release process, and the rule that `build-release.yml`, `pr-checks.yml` and `publish-stable.yml` stay frozen for `@main` callers. The README gains a Versioning section.

## Things to look at

- Shellcheck findings inside workflows are reported at warning severity and above. The frozen legacy workflows have 21 info and style findings (mostly SC2086), and fixing them means editing files every `@main` caller runs.
- `.github/actionlint.yaml` suppresses one real error: `github.job_workflow_sha` in `check-image-updates.yml` does not exist, so its script checkout always takes `main`. The fix and the removal of the suppression are [issue 58](https://github.com/halos-org/shared-workflows/issues/58).
- The plan named the file `lint.yml`; it is `ci.yml` because it also runs script tests.

## Verification

- `tests/move-major-tag.test.sh` runs the script against a throwaway bare origin: first release, newer release, rc, older patch release, second major, malformed tag. 8 of 8 pass locally.
- actionlint exits 0 on this branch. A workflow that references an undefined input makes it exit 1.
- The real tag move is exercised when `v1.0.0` is published ([issue 54](https://github.com/halos-org/shared-workflows/issues/54)); the rc before it must leave `v1` absent.

Outside the diff: the stale `probe.yml` and `guardprobe.yml` workflow registrations from deleted test branches are disabled. After merge, branch protection on `main` will require `status`.

Closes https://github.com/halos-org/shared-workflows/issues/50
Closes https://github.com/halos-org/shared-workflows/issues/41

---

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

- **New Features**
  - Automatically updates major version tags when a new full release is published.
  - Prevents prereleases and older releases from moving major tags.

- **Bug Fixes**
  - Improved workflow validation by checking GitHub Actions syntax, shell scripts, and repository tests.

- **Documentation**
  - Added guidance on release versioning, tag behavior, workflow compatibility, and recommended version pinning.
  - Added repository contribution and workflow usage guidance.

### CODE REVIEW

File: scripts/move-major-tag.sh [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+# the highest vX.*.* tag, so publishing a patch for an older minor leaves the
+# major tag on the newest release. Pre-release tags such as v1.0.0-rc.1 are
+# skipped. Any other tag name is an error.
+set -euo pipefail
+
+TAG=${1:?usage: move-major-tag.sh <release-tag>}
+RELEASE_RE='^v([0-9]+)\.[0-9]+\.[0-9]+$'
+PRERELEASE_RE='^v[0-9]+\.[0-9]+\.[0-9]+-'
+
+if [[ $TAG =~ $PRERELEASE_RE ]]; then
```
[the remark this answers is not in the document]
Comment (mairas) [reply scripts/move-major-tag.sh:16]: Fixed in d06d8b6. `PRERELEASE_RE` now requires a semver pre-release suffix: dot-separated identifiers of alphanumerics and hyphens, anchored at the end. `v1.0.0-` and `v1.0.0-rc_1` now exit non-zero. The malformed-tag test covers both and checks that `v1` stays absent.

**Comment by mairas** [comment]
Force-pushed to fold a fix into `feat: move the major tag on each full release` (now d06d8b6), from the CodeRabbit finding. `scripts/move-major-tag.sh` pre-release matching now requires a valid semver suffix, so `v1.0.0-` and `v1.0.0-rc_1` fail instead of being skipped as pre-releases. `tests/move-major-tag.test.sh` gains those cases: 11 of 11 pass locally, shellcheck clean.
